### PR TITLE
Automated releases - Round 2

### PR DESCRIPTION
**Pull request by pradyunsg**
Follow up from #265.

Based on experience trying to make the 20.2 release, as per #272.

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -215,3 +236,63 @@ def _bump(session, *, version, file, kind):
     session.log(f"git commit")
     subprocess.run(["git", "add", str(file)])
     subprocess.run(["git", "commit", "-m", f"Bump for {kind}"])
+
+
+@contextlib.contextmanager
+def _replace_file(original_path):
+    # Create a temporary file
```
Comment (brettcannon) [inline noxfile.py]:
```suggestion
    # Create a temporary file.
```

File: noxfile.py [hunk 2]
```diff
@@ -215,3 +236,63 @@ def _bump(session, *, version, file, kind):
...
+    title = f"{version} - {yyyy_mm_dd}"
+
+    with _replace_file(file) as (original, replacement):
+        for line in original:
+            if line == "*unreleased*\n":
+                replacement.write(f"{title}\n")
+                replacement.write(len(title) * "~" + "\n")
+                # Skip processing the next line (the heading underline for *unreleased*)
+                # since we already wrote the heading underline.
+                next(original)
```
Comment (brettcannon) [inline noxfile.py]: Worth adding an `assert` to check the assumption the comment is referencing?

File: noxfile.py [hunk 3]
```diff
@@ -215,3 +236,63 @@ def _bump(session, *, version, file, kind):
...
+            else:
+                replacement.write(line)
+
+
+def _changelog_add_unreleased_title(*, file):
+    with _replace_file(file) as (original, replacement):
+        # Duplicate first 3 lines from the original file.
```
Comment (brettcannon) [inline noxfile.py]: What's being duplicated? Is it worth putting it in this file as a template instead to make sure this assumption doesn't change?

File: noxfile.py (hunk 2 above)
Comment (pradyunsg) [reply noxfile.py]: Possibly. I'll do that in round 3.

File: noxfile.py (hunk 3 above)
Comment (pradyunsg) [reply noxfile.py]: Possibly. I'll explore in round 3.

File: docs/development/release-process.rst [hunk 4]
```diff
@@ -6,19 +6,14 @@ Release Process
 
     $ pip install nox
 
-#. Modify the ``CHANGELOG.rst`` to include changes made since the last release
```
Comment (brettcannon) [inline docs/development/release-process.rst]: Don't we still need to add the entries?

Comment (pradyunsg) [reply docs/development/release-process.rst]: I mean, I guess? We don't have to call this out in the process tho -- we should really be doing it on a per-PR basis.

Anyway, we can re-add in a follow up.

**Comment by pradyunsg** [comment]
I've gone ahead and merged this, since I have time rn to make the 20.2 release and this is a blocker for that. I don't think there were any blocker comments; and if there were, well, I missed them, sorry!

**Comment by brettcannon** [comment]
Nope, nothing really blocking this.

**Pull request merged**